eth/filters/api.go

```diff
@@ -345,6 +347,9 @@ func (api *PublicFilterAPI) GetLogs(ctx context.Context, crit FilterCriteria) ([
 		if crit.ToBlock != nil {
 			end = crit.ToBlock.Int64()
 		}
+		if end-begin+1 > api.maxBlockRange {
```

Comment:
can add a unit test here. e.g., `api.maxBlockRange` is 2, test fetch 2 (happy path) and fetch 3 (error path).